D06 - Protect Secrets.md

```diff
@@ -1,5 +1,27 @@
 # D06 - Protect Secrets
 
+Secrets provide access to resources or help encrypting data in transit or at
+rest. Secrets could be private SSH keys, private keys for a X509 certificate,
+HMACs for JWT or similar cryptograptographic material. Also it could be just
+secrets which provide access to something or just holds credentials, like backend
+connectors to databases, API keys, image registry keys or any other tokens which
+provide access to an internal or external service.
+
+Likely you cannot avoid always to provide some kind of keys or credentails to
```

Comment:
It might sound better like  :  "you cannot always avoid"
credentails can be changed to credentials